File: Emulators/EmulatorProfileRegistry.cs
using PointerFinder2.Emulators.EmulatorManager;
using PointerFinder2.Emulators.LiveScan.Dolphin;
using PointerFinder2.Emulators.LiveScan.DuckStation;
using PointerFinder2.Emulators.LiveScan.PCSX2;
// Added using statements for the new PPSSPP scanner strategies.
using PointerFinder2.Emulators.LiveScan.PPSSPP;
using PointerFinder2.Emulators.LiveScan.RALibretro;
using PointerFinder2.Emulators.StateBased.Dolphin;
using PointerFinder2.Emulators.StateBased.DuckStation;
using PointerFinder2.Emulators.StateBased.PCSX2;
using PointerFinder2.Emulators.StateBased.PPSSPP;
using PointerFinder2.Emulators.StateBased.RALibretro;
// Added using statement for the static range finder forms.
using PointerFinder2.UI.StaticRangeFinders;
using System;
using System.Collections.Generic;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace PointerFinder2.Emulators
{
    // A central registry for all supported emulator profiles.
    // This is the main place to add or modify emulator support. To add a new
    // emulator, create its Manager and Scanner classes, then add a new
    // EmulatorProfile entry to this list.
    public static class EmulatorProfileRegistry
    {
        public static readonly List<EmulatorProfile> Profiles = new List<EmulatorProfile>
        {
            // Specify that this profile supports a static range finder and provide its factory.
            new EmulatorProfile
            {
                Name = "PCSX2",
                Target = EmulatorTarget.PCSX2,
                ProcessNames = new[] { "pcsx2-qt", "pcsx2" },
                ManagerFactory = () => new Pcsx2Manager(),
                ScannerFactory = () => new Pcsx2ScannerStrategy(),
                StateBasedScannerFactory = () => new Pcsx2StateBasedScannerStrategy(),
                SupportsStaticRangeFinder = true,
                StaticRangeFinderFactory = (manager) => new Pcsx2RamScanRangeFinderForm(manager)
            },
            new EmulatorProfile
            {
                Name = "DuckStation",
                Target = EmulatorTarget.DuckStation,
                ProcessNames = new[] { "duckstation-qt-x64-ReleaseLTCG" },
                ManagerFactory = () => new DuckStationManager(),
                ScannerFactory = () => new DuckStationScannerStrategy(),
                StateBasedScannerFactory = () => new DuckStationStateBasedScannerStrategy(),
                SupportsStaticRangeFinder = false
            },
            new EmulatorProfile
            {
                Name = "RALibretro (NDS)",
                Target = EmulatorTarget.RALibretroNDS,
                ProcessNames = new[] { "RALibretro" },
                ManagerFactory = () => new RALibretroNDSManager(),
                ScannerFactory = () => new RALibretroNDSScannerStrategy(),
                StateBasedScannerFactory = () => new RALibretroNDSStateBasedScannerStrategy(),
                SupportsStaticRangeFinder = true,
                StaticRangeFinderFactory = (manager) => new NdsStaticRangeFinderForm(manager)
            },
            // Added the new profile for Dolphin.
            new EmulatorProfile
            {
                Name = "Dolphin",
                Target = EmulatorTarget.Dolphin,
                ProcessNames = new[] { "Dolphin" },
                ManagerFactory = () => new DolphinManager(),
                ScannerFactory = () => new DolphinLiveScannerStrategy(),
                StateBasedScannerFactory = () => new DolphinStateBasedScannerStrategy(),
                SupportsStaticRangeFinder = true,
                StaticRangeFinderFactory = (manager) => new DolphinFileRangeFinderForm(manager)
            },
            // Added the new profile for PPSSPP.
            new EmulatorProfile
            {
                Name = "PPSSPP",
                Target = EmulatorTarget.PPSSPP,
                ProcessNames = new[] { "PPSSPPWindows64" },
                ManagerFactory = () => new PpssppManager(),
                ScannerFactory = () => new PpssppLiveScannerStrategy(),
                StateBasedScannerFactory = () => new PpssppStateBasedScannerStrategy(),
                SupportsStaticRangeFinder = true,
                StaticRangeFinderFactory = (manager) => new PpssppRamScanRangeFinderForm(manager)
            }
        };
    }
}

File: src/Emulators/EmulatorProfileRegistry.cpp
#include "EmulatorProfileRegistry.h"

#include "Managers/DolphinManager.h"
#include "Managers/DuckStationManager.h"
#include "Managers/Pcsx2Manager.h"
#include "Managers/PpssppManager.h"
#include "Managers/RALibretroNDSManager.h"
#include "Managers/RALibretroGBAManager.h"
#include "Strategies/DolphinStateBasedScanner.h"
#include "Strategies/DuckStationStateBasedScanner.h"
#include "Strategies/Pcsx2StateBasedScanner.h"
#include "Strategies/PpssppStateBasedScanner.h"
#include "Strategies/RALibretroNDSStateBasedScanner.h"
#include "Strategies/RALibretroGBAStateBasedScanner.h"

namespace PointerFinder2::Emulators {

    using namespace PointerFinder2::DataModels;
    using namespace PointerFinder2::Emulators::EmulatorManager;
    using namespace PointerFinder2::Emulators::StateBased;

    // Returns the master registry list, detailing process names and custom scanning 
    // procedures for each supported system (PCSX2, DuckStation, Dolphin, PPSSPP, and RALibretro).
    const std::vector<EmulatorProfile>& EmulatorProfileRegistry::getProfiles() {
        static const std::vector<EmulatorProfile> profiles = {
            {
                "PCSX2",
                EmulatorTarget::PCSX2,
                { "pcsx2-qt", "pcsx2" },
                []() { return std::make_unique<Pcsx2Manager>(); },
                []() { return std::make_unique<Pcsx2StateBasedScanner>(); },
                true
            },
            {
                "DuckStation",
                EmulatorTarget::DuckStation,
                { "duckstation-qt-x64-ReleaseLTCG" },
                []() { return std::make_unique<DuckStationManager>(); },
                []() { return std::make_unique<DuckStationStateBasedScanner>(); },
                false
            },
            {
                "RALibretro (NDS)",
                EmulatorTarget::RALibretroNDS,
                { "RALibretro" },
                []() { return std::make_unique<RALibretroNDSManager>(); },
                []() { return std::make_unique<RALibretroNDSStateBasedScanner>(); },
                true
            },
            {
                "Dolphin",
                EmulatorTarget::Dolphin,
                { "Dolphin" },
                []() { return std::make_unique<DolphinManager>(); },
                []() { return std::make_unique<DolphinStateBasedScanner>(); },
                true
            },
            {
                "PPSSPP",
                EmulatorTarget::PPSSPP,
                { "PPSSPPWindows64" },
                []() { return std::make_unique<PpssppManager>(); },
                []() { return std::make_unique<PpssppStateBasedScanner>(); },
                true
            },
            {
                "RALibretro (GBA)",
                EmulatorTarget::RALibretroGBA,
                { "RALibretro" },
                []() { return std::make_unique<RALibretroGBAManager>(); },
                []() { return std::make_unique<RALibretroGBAStateBasedScanner>(); },
                false
            }
        };
        return profiles;
    }

}